vision_evaluation/evaluators.py

```diff
@@ -816,3 +817,233 @@ def __init__(self):
+        self.num_samples = 0
+        self.metric_sum = 0
+
+    def add_predictions(self, predictions, targets):
+        """ Adding predictions and ground truth of images for image matting task
+        Args:
+            predictions: list of image matting predictions, [matting1, matting2, ...]. Shape: (N, ), type: PIL image object or Numpy array
+            targets: list of image matting ground truth, [gt1, gt2, ...]. Shape: (N, ), type: PIL image object or Numpy array
+        """
+        gt_mask_list = []
+        pred_mask_list = []
+        for pred_mask, gt_mask in zip(predictions, targets):
+            pred_mask = np.asarray(pred_mask)
+            gt_mask = np.asarray(gt_mask)
+
+            pred_binmask = self._convert2binary(pred_mask)
+            gt_binmask = self._convert2binary(gt_mask)
+            gt_boundary_mask, pred_boundary_mask = self._create_contour_mask(gt_binmask, pred_binmask)
+            gt_mask_list.append(gt_boundary_mask.astype(np.int64))
+            pred_mask_list.append(pred_boundary_mask.astype(np.int64))
+        self.base_evaluator.add_predictions(pred_mask_list, gt_mask_list, convert_to_binary=False)
+
+    def get_report(self):
+        return {self.metric: self.base_evaluator.metric_sum / self.base_evaluator.num_samples if self.base_evaluator.num_samples else 0.0}
+
+
+class BoundaryForegroundIOUEvaluator(MattingEvaluatorBase):
+    """
+    Boundary foreground intersection-over-union evaluator
+    """
+    def __init__(self):
+        super(BoundaryForegroundIOUEvaluator, self).__init__()
+        self.metric = 'b_fgIOU'
+        self.base_evaluator = ForegroundIOUEvaluator()
+        self.num_samples = 0
+        self.metric_sum = 0
+
+    def add_predictions(self, predictions, targets):
+        """ Adding predictions and ground truth of images for image matting task
+        Args:
+            predictions: list of image matting predictions, [matting1, matting2, ...]. Shape: (N, ), type: PIL image object or Numpy array
+            targets: list of image matting ground truth, [gt1, gt2, ...]. Shape: (N, ), type: PIL image object or Numpy array
+        """
+        gt_mask_list = []
+        pred_mask_list = []
+        for pred_mask, gt_mask in zip(predictions, targets):
+            pred_mask = np.asarray(pred_mask)
+            gt_mask = np.asarray(gt_mask)
+
+            pred_binmask = self._convert2binary(pred_mask)
+            gt_binmask = self._convert2binary(gt_mask)
+            gt_boundary_mask, pred_boundary_mask = self._create_contour_mask(gt_binmask, pred_binmask)
+            gt_mask_list.append(gt_boundary_mask.astype(np.int64))
+            pred_mask_list.append(pred_boundary_mask.astype(np.int64))
+        self.base_evaluator.add_predictions(pred_mask_list, gt_mask_list, convert_to_binary=False)
+
+    def get_report(self):
+        return {self.metric: self.base_evaluator.metric_sum / self.base_evaluator.num_samples if self.base_evaluator.num_samples else 0.0}
```

Comment:
is this just {self.metric: self.base_evaluator.get_report()[self.base_evaluator.metric_name]}?